utils/src/main/kotlin/LicenseFilenamePatterns.kt

```diff
@@ -44,4 +46,66 @@ object LicenseFilenamePatterns {
     val ROOT_LICENSE_FILENAMES = listOf(
         "readme*"
     ).generateCapitalizationVariants()
+
+    /**
+     * Return glob patterns which match all files which may contain license information residing recursively within the
+     * given absolute [directory] or in any of its ancestor directories.
+     */
+    fun getLicenseFileGlobsForDirectory(directory: String): List<String> =
+        globFileInDirectoryOrAncstors(directory, LICENSE_FILENAMES + ROOT_LICENSE_FILENAMES)
+
+    /**
+     * Return recursively all parent directories of the given relative [directory]. An empty [String] in the result
```

Comment:
"parent" -> "ancestor" to align with the function name.

Reply:
Done.